app/src/androidTest/java/com/google/samples/apps/topeka/activity/SignInActivityTest.java

```diff
@@ -65,15 +71,33 @@ public void checkFab_initiallyNotDisplayed() {
     }
 
     @Test
+    public void signIn_checkIfEditTextViewsAreInitiallyEmpty() {
+        onView(withId(R.id.first_name)).check(matches(withText(isEmptyOrNullString())));
+        onView(withId(R.id.last_initial)).check(matches(withText(isEmptyOrNullString())));
+    }
+
+    @Test
     public void signIn_performSuccessful() {
         inputData();
-        onView(withId(R.id.done)).check(matches(isDisplayed()));
+        onView(withId(R.id.done)).check(matches(isDisplayed())).perform(click());
+        onView(withId(R.id.toolbar_player)).check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void signIn_chooseAvatar() {
+        for (int i = 0; i < 16; i++) {
+            onData(anything())
+                    .inAdapterView(withId(R.id.avatars))
+                    .atPosition(15)
+                    .check(matches(isDisplayed()))
```

Comment:
@ardock There's a performance penalty that has to be paid if you check the column index rather than hard coding these values. This is why I went with the route of documenting the use and origin of these numbers.